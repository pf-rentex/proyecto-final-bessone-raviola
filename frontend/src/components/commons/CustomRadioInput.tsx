import React, {useEffect, useState} from 'react';

interface ICustomCheckboxProps {
  id: string;
  name: string;
  isChecked: boolean;
  onSelect: Function;
  text: string;
}

const CustomRadioInput = ({id, name, isChecked, onSelect, text}: ICustomCheckboxProps) => {

  return (
      <div onClick={() => onSelect()}
          className={`rounded-lg flex w-36 h-10 px-2 py-2 my-2 mx-2 ${isChecked ? 'bg-blue-500 shadow-inner-xl': 'bg-gray-300 shadow-lg' } place-content-center cursor-pointer hover:opacity-75`}>
        <input type="radio"
               name={name}
               className="hidden w-0"
               id={id}
               value={id}
               onChange={() => onSelect()}
               checked={isChecked}
               hidden/>
        <span className={`select-none radio text-left text-xs leading-4 font-bold ${isChecked ? 'text-gray-100': 'text-gray-900' } self-center text-start rounded-lg`}>
          { text }
        </span>
      </div>
  );
};

export default CustomRadioInput;
