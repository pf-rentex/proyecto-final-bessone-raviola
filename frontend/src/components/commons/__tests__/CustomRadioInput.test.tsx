import React from 'react';
import {render} from '@testing-library/react';
import CustomRadioInput from '../CustomRadioInput';

describe('Custom Radio Input', () => {
  test('renders button', () => {
    const component = render(<CustomRadioInput name="test"
                                               text="radio"
                                               id="id"
                                               isChecked={true}
                                               onSelect={() => {}}
    />);

    expect(component.container).toBeInTheDocument();
    expect(component.container).toMatchSnapshot();
  });

  test('toggles button', async () => {
    let checked = false;
    let onCheck = jest.fn(() => checked = true);
    const {getByLabelText, rerender } = await render(<CustomRadioInput name="test"
                                                                      text="radio"
                                                                      id="id"
                                                                      isChecked={checked}
                                                                      onSelect={onCheck}
    />);

    const input = getByLabelText('radio');

    expect(input).not.toBeChecked();

    await input.click()

    expect(onCheck).toHaveBeenCalled();
    expect(checked).toBeTruthy();

    rerender(<CustomRadioInput name="test"
                               text="radio"
                               id="id"
                               isChecked={checked}
                               onSelect={onCheck}
    />);
    expect(input).toBeChecked();

  });

});
