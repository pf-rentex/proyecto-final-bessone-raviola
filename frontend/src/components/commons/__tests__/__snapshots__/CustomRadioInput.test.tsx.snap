// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Radio Input renders button 1`] = `
<div>
  <div
    class="rounded-lg flex w-32 h-12 px-2 py-2 my-2 mx-2 bg-blue-500 shadow-inner-xl place-content-center cursor-pointer hover:opacity-75"
  >
    <label>
      <input
        checked=""
        class="hidden w-0"
        hidden=""
        id="id"
        name="test"
        type="radio"
        value="id"
      />
      <span
        class="select-none radio text-left text-xs leading-4 font-bold text-gray-100 self-center text-start rounded-lg"
      >
        radio
      </span>
    </label>
  </div>
</div>
`;
